lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -15,22 +15,20 @@ module GitHelper
       #         (i.e. a local working copy) or a subdirectory of one.
       #
       def self.is_git_repo?(path: nil)
-        args = ['git']
-
-        unless path.nil?
-          # `path` might represent a file that doesn't yet exist but is in a
-          # valid Git repo. In such case, the `git rev-parse` we're currently
-          # building would fail. To avoid that, let's perform the check on the
-          # first existing ancestor of the given path – which will be the path
-          # itself if it exists.
-          path = first_existing_ancestor_of(path: path)
-          args += ['-C', File.file?(path) ? File.dirname(path) : path]
-        end
+        working_path = path.nil? ? Dir.pwd : first_existing_ancestor_of(path: path)
+        current_dir = File.directory?(working_path) ? working_path : File.dirname(working_path)
 
-        args += ['rev-parse']
+        # Calling `expand_path` to get an absolute path and avoid and endless
+        # loop of adding ".." to the path without ever triggerting the "/"
+        # safeguard.
+        full_path = Pathname(current_dir).expand_path
 
-        Action.sh(args, print_command_output: false) do |status, _, _|
-          status.success?
+        if full_path.children.map { |f| File.basename(f) }.include?('.git')
+          return true
+        elsif full_path == ROOT_FOLDER
+          return false
+        else
+          return is_git_repo?(path: full_path.parent)
```

Comment:
I'm not too happy with how this code _looks_.

Keen to hear if you have any suggestions 😄 

Also notice that I added two extra tests for the recursively traversing scenario in the base branch: 6cf529fa2c45ff87051da79c1d40a206e55a0d73

Reply:
Feels a little dense in the first two lines – I don't think I'd expand out the ternaries, but maybe a well-placed comment or two would help it flow a bit more nicely? 🤔 

IMHO not a huge deal though – grokking it only took a few seconds 🤷‍♂️

Reply:
One thing that also feels strange with this is that there are 2 loops / recursion levels: one within `first_existing_ancestor_of` to go up until it finds an existing ancestor, and another one here to go up until it finds a `.git` or the root folder. Couldn't we avoid going thru the `first_existing_ancestor_of` loop on subsequent recursion loops, to avoid unnecessarily repeating some steps there?

Personally I think it's not worth using recursion here as opposed to using a `while` or `until` loop, something like:

```ruby
until (full_path + '.git').exist? || full_path.root? do
  full_path = full_path.parent
end
```